### PR TITLE
Unit test assembly for our MSBuild tasks

### PR DESCRIPTION
**Pull request by jaredpar**
This is the start of our unit test assembly for our MSBuild assets.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @agocke, @khyperia, @VSadov

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 1]
```diff
@@ -479,6 +479,11 @@ private void LogErrorOutput(string output)
         /// </remarks>
         internal abstract void LogMessages(string output, MessageImportance messageImportance);
 
+        public string GenerateResponseFileContents()
```
Comment (khyperia) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: Is there a reason this is named differently than what it's calling through to?

**Comment by khyperia** [comment]
Other than that minor detail, LGTM

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: @khyperia the method it's calling is `protected` so it's not directly accessible. I thought about shadowing the name but it seemed to cause more confusion that I wanted.  Open to suggestions for better names.

**Comment by tmeschter** [comment]
LGTM

**Pull request merged**